src/promptflow/tests/executor/unittests/_utils/test_dataclass_serializer.py

```diff
@@ -0,0 +1,204 @@
+                "api_base": "https://gpt-test-eus.openai.azure.com/",
+                "api_type": "azure",
+                "api_version": "2023-07-01-preview",
+            },
+        },
+        "custom_connection": {
+            "type": "CustomConnection",
+            "value": {
+                "api_key": "<your-key>",
+                "url": "https://api.bing.microsoft.com/v7.0/search",
+            },
+            "module": "promptflow.connections",
+            "secret_keys": ["api_key"],
+        },
+    }
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    "type_input, expected",
+    [
+        (NodeRunRecord, NodeRunRecord),
+        ([NodeRunRecord], List[NodeRunRecord]),
+        (dict(a=NodeRunRecord), Dict[str, NodeRunRecord]),
+        (int, int),
+        (str, str),
+    ]
+)
+def test_get_type(type_input, expected):
+    assert get_type(type_input) == expected
+
+
+@pytest.mark.unittest
+def test_serialize_dataclass():
+    start_time = datetime(2023, 9, 4)
+    end_time = datetime(2023, 9, 4)
+    node_run_info = RunInfo(
+        node=None,
+        run_id=None,
+        flow_run_id=None,
+        status=Status.Completed,
+        inputs=None,
+        output=None,
+        metrics=None,
+        error=None,
+        parent_run_id=None,
+        start_time=start_time,
+        end_time=end_time,
+        index=0,
+    )
+    node_record = NodeRunRecord.from_run_info(node_run_info)
+    serialized_run_info = serialize(node_run_info)
+    deserialized_run_info = deserialize_value(serialized_run_info, RunInfo)
+    assert deserialized_run_info == node_run_info
+    assert serialize(node_record) == node_record.serialize()
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    "value, value_type",
+    [
+        (datetime(2023, 9, 4), datetime),
+        (Status.Completed, Status),
+        ([1, 2, 3], List[int]),
+        ({"a": 1, "b": 2}, Dict[str, int]),
+        (1, int),
+        ("a", str),
+    ]
+)
+def test_serialize_value(value, value_type):
+    assert deserialize_value(serialize(value), value_type) == value
+
+
+@pytest.mark.unittest
+def test_serialize_remove_null():
+    value = {"a": 1, "b": None}
+    value_type = Dict[str, int]
+    assert deserialize_value(serialize(value, remove_null=True), value_type) == {"a": 1, "b": None}
+
+    @dataclass
+    class DummyDataClass:
+        name: str
+        age: int
+    serialized_dataclass = serialize(DummyDataClass("Dummy", None), remove_null=True)
+    assert serialize(serialized_dataclass, remove_null=True) == {'name': 'Dummy'}
+
+    new_connection = get_connection_dict()
+    connection_manager = ConnectionManager(new_connection)
+    assert serialize(connection_manager.get("azure_open_ai_connection"), remove_null=True) == "azure_open_ai_connection"
+
+    def generator():
+        for i in range(3):
+            yield i
+    g = GeneratorProxy(generator())
+    next(g)
+    assert serialize(g, remove_null=True) == [0]
```

Comment:
Why this generator test is not in a new test case?

Reply:
Moved this test to a new function test_serialize_generator().